types/stylelint/index.d.ts

```diff
@@ -7,7 +7,10 @@ declare module 'stylelint' {
 		export type Severity = 'warning' | 'error';
 
 		export type ConfigExtends = string | string[];
-		export type ConfigPlugins = string | string[];
+		export type PluginImportType =
+			| { default?: { ruleName: string; rule: Rule } }
+			| { ruleName: string; rule: Rule };
+		export type ConfigPlugins = string | string[] | PluginImportType | PluginImportType[];
```

Comment:
[suggest] Since the existing `Plugin` type name is inaccurate, I believe we should use the type in different semantics.

So could you please fix the types like this? I confirm the type-check passes locally.

```diff
@@ -6,11 +6,14 @@ declare module 'stylelint' {
 	namespace stylelint {
 		export type Severity = 'warning' | 'error';
 
+		// TODO: This type will be renamed to `Plugin` with the next major version.
+		export type PluginType = {
+			ruleName: string;
+			rule: Rule;
+		};
+
 		export type ConfigExtends = string | string[];
-		export type PluginImportType =
-			| { default?: { ruleName: string; rule: Rule } }
-			| { ruleName: string; rule: Rule };
-		export type ConfigPlugins = string | string[] | PluginImportType | PluginImportType[];
+		export type ConfigPlugins = string | PluginType | Array<string | PluginType>;
 		export type ConfigProcessor = string | [string, Object];
 		export type ConfigProcessors = string | ConfigProcessor[];
 		export type ConfigIgnoreFiles = string | string[];
@@ -207,6 +210,7 @@ declare module 'stylelint' {
 			meta?: RuleMeta;
 		};
 
+		// TODO: This type will change semantically with the next major version. See `PluginType`.
 		export type Plugin = RuleBase;
 
 		export type CodeProcessor = (code: string, file: string | undefined) => string;
@@ -466,7 +470,7 @@ declare module 'stylelint' {
 			/**
 			 * Creates a Stylelint plugin.
 			 */
-			createPlugin: (ruleName: string, rule: Rule) => { ruleName: string; rule: Rule };
+			createPlugin: (ruleName: string, rule: Rule) => PluginType;
 			/**
 			 * Internal use only. Do not use or rely on this method. It may
 			 * change at any time.
```

Reply:
Thanks, @ybiquitous 
I fixed the linting errors 🙇🏽